ibis/backends/pyspark/__init__.py

```diff
@@ -101,12 +100,15 @@ class Options(ibis.config.BaseModel):
     def _from_url(self, url: str) -> Backend:
         """Construct a PySpark backend from a URL `url`."""
         url = sa.engine.make_url(url)
-        params = list(itertools.chain.from_iterable(url.query.items()))
+        params = list(url.query.items())
         if database := url.database:
-            params.append("spark.sql.warehouse.dir")
-            params.append(str(Path(database).absolute()))
+            params.append(
+                ("spark.sql.warehouse.dir", str(Path(database).absolute()))
+            )
 
-        builder = SparkSession.builder.config(*params)
+        builder = SparkSession.builder
+        for key, value in params:
```

Comment:
How about directly calling `builder.config()` instead of having an intermediate params container?

Reply:
You can't do that, `.config` takes a single key value pair. It's meant to be chained.

Reply:
Ah, but I can use `SparkConf.setAll`!